sstables/storage.cc

```diff
@@ -367,15 +405,15 @@ future<> filesystem_storage::move(const sstable& sst, sstring new_dir, generatio
     sstlog.debug("Moving {} old_generation={} to {} new_generation={} do_sync_dirs={}",
             sst.get_filename(), sst._generation, new_dir, new_generation, delay_commit == nullptr);
     co_await create_links_common(sst, new_dir, new_generation, mark_for_removal::yes);
-    _dir = new_dir;
+    co_await change_dir(new_dir);
```

Comment:
With `open_direcectory::chdir` method, this could be `co_await _dir.chdir(new_dir)`
I think it'd be clearer to reader to understand what's going here this way.

Reply:
I'm skipping this for now; per Avi's [comment](https://github.com/scylladb/scylladb/pull/19624#discussion_r1683286215).